beacon-chain/state/stateV0/state_trie.go

```diff
@@ -18,14 +19,22 @@ import (
 	"github.com/prysmaticlabs/prysm/shared/params"
 	"github.com/prysmaticlabs/prysm/shared/sliceutil"
 	"go.opencensus.io/trace"
+	"google.golang.org/protobuf/proto"
 )
 
 // InitializeFromProto the beacon state from a protobuf representation.
 func InitializeFromProto(st *pbp2p.BeaconState) (*BeaconState, error) {
-	return InitializeFromProtoUnsafe(proto.Clone(st).(*pbp2p.BeaconState))
+	return InitializeFromProtoUnsafe(hydrateState(proto.Clone(st).(*pbp2p.BeaconState)))
+}
+
+func hydrateState(st *pbp2p.BeaconState) *pbp2p.BeaconState {
+	if st.Validators == nil {
+		st.Validators = []*ethpb.Validator{}
+	}
+	return st
 }
```

Comment:
Added this because empty arrays after performing a proto.Clone would be nil. Wanted to make sure this is a valid workaround.